### PR TITLE
Mail block

### PR DESCRIPTION
**Pull request by sjrodahl**
Simple mail sender block that uses no authentication and sends to a local, fake SMTP-server running on host.

### CODE REVIEW

**Comment by Lagostra** [comment]
The fakeSMTP program should be added to dockerignore to avoid inflating build times

File: src/blocks/send_mail.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        msg.set_content(params['message'])
+
+        # Switch to appropriate mail server address.
+        try:
+            with smtplib.SMTP('172.17.0.1', 12000) as smtp:
+                smtp.send_message(msg)
+            return {'email_sent': True}
+        #TODO::Fix error handling to better know what went wrong
+        except:
+            return {'email_sent': False}
```
[the remark this answers is not in the document]
Comment (sjrodahl) [reply src/blocks/send_mail.py]: Isn't it better if this is handled by main?

[the remark this answers is not in the document]
Comment (Lagostra) [reply src/blocks/send_mail.py]: True - it should raise an Exception, and then main should catch it and send a 400.

[the remark this answers is not in the document]
Comment (Lagostra) [reply src/blocks/send_mail.py]: Or a 500. We should actually probably have defined a set of exceptions that can be raised, and respond accordingly, but I don't think that is a priority...

File: src/blocks/send_mail.py [hunk 2]
```diff
@@ -46,25 +47,23 @@ def get_info(self):
...
-        except smtplib.SMTPAuthenticationError as auth_e:
-            print('Username and password not accepted.\n' + auth_e.message())
-            return {'email_sent': False}
-        except smtplib.SMPTException as e:
-            print('An error occured: ' + e.message())
+        except smtplib.SMTPConnectError:
+            raise BlockError("Unable to connect to the SMTP server {}".format(server_name))
+        except smtplib.SMTPAuthenticationError:
+            raise BlockError('Username and password not accepted to {}'.format(server_name))
+        except smtplib.SMPTException:
```
Comment (Lagostra) [inline src/blocks/send_mail.py]: This appears to be a typo (SMPTException -> SMTPException)

File: src/main.py [hunk 3]
```diff
@@ -76,7 +78,11 @@ def execute_block(block_name):
     if is_resuming:
         result = block.resume(body['state'], cleaned_params)
```
Comment (Lagostra) [inline src/main.py]: Should catch exceptions from this as well, so just wrap the try around the entire if-else structure

File: src/blocks/send_mail.py (hunk 2 above)
Comment (sjrodahl) [reply src/blocks/send_mail.py]: Good catch!

**Pull request merged**